pkg/kubelet/kubelet_pods.go

```diff
@@ -1641,12 +1640,9 @@ func (kl *Kubelet) determinePodResizeStatus(pod *v1.Pod, podStatus *v1.PodStatus
 		if err := kl.statusManager.SetPodResizeStatus(pod.UID, ""); err != nil {
 			klog.ErrorS(err, "SetPodResizeStatus failed", "pod", pod.Name)
 		}
-	} else {
-		if resizeStatus, found := kl.statusManager.GetPodResizeStatus(string(pod.UID)); found {
-			podResizeStatus = resizeStatus
-		}
+		return ""
```

Comment:
em. I doubt external loop check empty here. could you double check it? We probably need some default/original status here.

Reply:
this one was added by @LastNight1997 
Shengjie, can you take a look?

Reply:
we get default/original status from pod.Status.Resize instead of local statusManager checkpoint, because if pod resource is updated , the resize status will be changed to Proposed, then the status cached in statusManager will be outdated. This will cause some problem, for example, the pod will stuck in Proposed when we update deferred pod to deferred again. And I also suggest that we can do away with local checkpoint and rely on PodStatus as persistent store and source of truth for resizeStatus and allocated resources, as @vinaykul said in https://github.com/kubernetes/kubernetes/issues/117589#issuecomment-1533982936.

Reply:
I see, I'm adding the ut to try to cover Proposed case